src/plugins/kibana_usage_collection/server/collectors/cloud/detector/aws.test.ts

```diff
@@ -6,120 +6,120 @@
-
-const AWS = new AWSCloudService();
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { readFile } = require('fs/promises') as { readFile: jest.Mock };
 
 describe('AWS', () => {
-  const expectedFilenames = ['/sys/hypervisor/uuid', '/sys/devices/virtual/dmi/id/product_uuid'];
-  const expectedEncoding = 'utf8';
-  // mixed case to ensure we check for ec2 after lowercasing
-  const ec2Uuid = 'eC2abcdef-ghijk\n';
-  const ec2FileSystem = {
-    readFile: (filename: string, encoding: string, callback: Callback) => {
-      expect(expectedFilenames).toContain(filename);
-      expect(encoding).toEqual(expectedEncoding);
-
-      callback(null, ec2Uuid);
-    },
-  } as typeof fs;
+  const mockIsWindows = jest.fn();
+  const awsService = new AWSCloudService();
+  awsService['_isWindows'] = mockIsWindows.mockReturnValue(false);
+  readFile.mockResolvedValue('eC2abcdef-ghijk\n');
 
+  beforeEach(() => jest.clearAllMocks());
   it('is named "aws"', () => {
-    expect(AWS.getName()).toEqual('aws');
+    expect(awsService.getName()).toEqual('aws');
   });
 
   describe('_checkIfService', () => {
     it('handles expected response', async () => {
       const id = 'abcdef';
-      const request = ((req: RequestOptions, callback: Callback) => {
-        expect(req.method).toEqual('GET');
-        expect(req.uri).toEqual(
-          'http://169.254.169.254/2016-09-02/dynamic/instance-identity/document'
-        );
-        expect(req.json).toEqual(true);
-
-        const body = `{"instanceId": "${id}","availabilityZone":"us-fake-2c", "imageId" : "ami-6df1e514"}`;
-
-        callback(null, { statusCode: 200, body });
-      }) as Request;
-      // ensure it does not use the fs to trump the body
-      const awsCheckedFileSystem = new AWSCloudService({
-        _fs: ec2FileSystem,
-        _isWindows: false,
+
+      fetchMock.mockResolvedValue({
+        json: () =>
+          `{"instanceId": "${id}","availabilityZone":"us-fake-2c", "imageId" : "ami-6df1e514"}`,
+        status: 200,
+        ok: true,
       });
 
-      const response = await awsCheckedFileSystem._checkIfService(request);
+      const response = await awsService['_checkIfService']();
+      expect(readFile).toBeCalledTimes(0);
+      expect(fetchMock).toBeCalledTimes(1);
+      expect(fetchMock).toBeCalledWith(
+        'http://169.254.169.254/2016-09-02/dynamic/instance-identity/document',
+        {
+          method: 'GET',
+        }
+      );
 
       expect(response.isConfirmed()).toEqual(true);
-      expect(response.toJSON()).toEqual({
-        name: AWS.getName(),
-        id,
-        region: undefined,
-        vm_type: undefined,
-        zone: 'us-fake-2c',
-        metadata: {
-          imageId: 'ami-6df1e514',
-        },
-      });
+      expect(response.toJSON()).toMatchInlineSnapshot(`
```

Comment:
Is there a reason to change from asserting equality to asserting against a snapshot? Snapshots tend to get stale and I don't see anything obvious that prevents us using `toEqual`.

Reply:
Both are equavilant in what they are checking. using inline snapshots saves us time manually changing the values inside the object rather than just updating the snapshots and verifying they adhere to the expected changes. I dont like using variables inside the `toEqual` from outside the thing that is being tested as it makes the test less verbose and might allow some bugs to pass.